woocommerce/woocommerce
# Added a woocommerce_disable_api_access_log filter.

### All Submissions:






### Changes proposed in this Pull Request:

Added a `woocommerce_disable_rest_api_access_log` filter to disable last access logging for rest api. Detail discussion concerning this PR can be found in issue #37144.

Closes #37144 .



### How to test the changes in this Pull Request:




1. Create a test plugin at wp-content/plugins/test/test.php
2. It should contain the following code:

```
<?php
/**
 * Plugin name: Test Code
 * Description: Test plugin.
 */

add_filter( 'woocommerce_disable_rest_api_access_log', function( $result, $user_id ) {
	return true;
}, 10, 2 );
```

3. Ensure woocommerce is active disable everything else, and then go ahead and activate Test Code Plugin.
4. Check the WooCommerce -> Settings -> Advanced -> Rest API.
5. Create new REST API key.
6. Note down Last access date.
7. Perform WooCommerce REST API call.
8. Observe the Last access date for key you have created.



### Other information:




### FOR PR REVIEWER ONLY:
